components/brave_wallet/browser/json_rpc_service_unittest.cc

```diff
@@ -3657,19 +3657,54 @@ TEST_F(JsonRpcServiceUnitTest, GetWalletAddrInvalidDomain) {
   }
 }
 
-TEST_F(JsonRpcServiceUnitTest, IsValidDomain) {
+TEST_F(JsonRpcServiceUnitTest, IsValidEnsDomain) {
   std::vector<std::string> valid_domains = {"brave.eth", "test.brave.eth",
                                             "brave-test.test-dev.eth"};
   for (const auto& domain : valid_domains) {
-    EXPECT_TRUE(JsonRpcService::IsValidDomain(domain))
+    EXPECT_TRUE(JsonRpcService::IsValidEnsDomain(domain))
         << domain << " should be valid";
   }
 
   std::vector<std::string> invalid_domains = {
       "",      ".eth",    "-brave.eth",      "brave-.eth",     "brave.e-th",
       "b.eth", "brave.e", "-brave.test.eth", "brave-.test.eth"};
   for (const auto& domain : invalid_domains) {
-    EXPECT_FALSE(JsonRpcService::IsValidDomain(domain))
+    EXPECT_FALSE(JsonRpcService::IsValidEnsDomain(domain))
+        << domain << " should be invalid";
+  }
+}
+
+TEST_F(JsonRpcServiceUnitTest, IsValidSnsDomain) {
+  std::vector<std::string> valid_domains = {
+      "brave.sol",                //
+      "test.brave.sol",           //
+      "brave-test.test-dev.sol",  //
+      "b.sol",                    //
+      "w.sol",                    //
+      "-.sol",                    //
+      "-brave.sol",               //
+      "brave-.sol",               //
+      "---.sol",                  //
```

Comment:
@kdenhartog 
Is it ok to support resolving of such hostnames like `-.sol`?
Such names are valid from their side https://www.sns.id/domain?domain=-----

Reply:
I'll need to double check the URL spec at WHATWG. Do all of these correctly validate with `GURL::is_valid()`? That would be my primary concern and then making sure the domains match up with URL spec and RFC3986 would be my secondary concerns.

Reply:
Well https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2 says 
```
Such a name consists of a sequence of domain labels separated by ".", each domain label starting and ending with an alphanumeric character and possibly also containing "-" characters.
```
This is ok
```
EXPECT_TRUE(GURL("https://-.sol").is_valid());
```
This also works
![image](https://github.com/brave/brave-core/assets/168356/e706ec38-3439-4a92-b9d2-3a868d27fd05)

Reply:
Ok, thanks for confirming. After looking through this further this makes sense. There's some similar cases when dealing with punycode conversions from emojis and looking at the definition, the host is non-empty and it's using valid ASCII code points so this checks out to me.